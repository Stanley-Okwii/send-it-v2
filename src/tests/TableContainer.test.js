import React from "react";
import { shallow, mount } from "enzyme";

import {
    TableContainer as App,
  mapStateToProps,
  mapDispatchToProps
} from "../containers/TableContainer";
import { getAllParcels } from "./mock-data";


describe("TableContainer ", () => {
  it("renders without crashing", () => {
      const wrapper = shallow(<App fetchParcels={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches state to props", () => {
    const initialState = {};
    mapStateToProps(initialState);
  });

  it("match dispatch to props", () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchParcels();
  });
});
