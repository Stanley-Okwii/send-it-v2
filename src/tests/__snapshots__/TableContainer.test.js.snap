// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainer  renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableContainer
    fetchParcels={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
